model/src/bruteforce.spthy

```diff
@@ -0,0 +1,103 @@
+// Brute-force Attacks:
+// - Allows the adversary to deduce cryptographically weak fresh values (human generated passwords).
+// - Brute-force(~v, x) allows the adversary to extract ~v if the adversary *might* know x.
+// - Deconstruction rules need not be considered because the adversary can apply them on their own before attempting to brute-force.
+//   However, we must ensure that the adversary can also brute-force those new targets.
+
+
+// Simplifications:
+// - The "target" for the brute-force attack must be deduced from the messages that are sent over the channel.
+// - Only the messages that are sent by the client must be considered.
+//   - The server's response in the first reconstruction phase adds a factor, which makes a brute-force attack more difficult (the adversary must know yet another factor!)
+//   - The server's response in the second reconstruction phase was previously sent over the channel. The adversary might as well brute-force it there already.
+// - The adversary need not have knowledge of the brute-force target, it suffices that the message was sent over the channel.
+
+
+// Adversary can brute-force a human generated value if the "weakness" propagates all the way to a target that the adversary might know
+rule bruteforce: // !HumanGenerated(~v) is repeated here to remind Tamarin that only human generated values can be brute-forced.
+    [ !HumanGenerated(~v), !BruteforceCandidate(t), !Bruteforce(~v, t) ]
+  --[ AdversaryAction()
+    , _restrict(~v << t)
+    , BruteforceAttack(~v, t) ]->
+    [ Out(~v) ]
+
+
+// Base Case: If ~v is known, extracting ~v is trivial
+rule bruteforce_hg: // considers only fresh values because human generated values must be fresh
+    [ !HumanGenerated(~v) ]-->[ !Bruteforce(~v, ~v) ]
+
+// pair
+rule bruteforce_pair: // combine two brute-force attacks: iterate over candidates (~v, ~w) and verify against <x, y> 
+    [ !Bruteforce(~v, x), !Bruteforce(~w, y) ]-->[ !Bruteforce(~v, <x, y>), !Bruteforce(~w, <x, y>) ]
```

Comment:
Why are `~v` and `~w` potentially different? I would have expected those two values to be the same.

Reply:
Aha. I think I see it. Because you maybe could brute force two values at the same time? Is this something we want to consider? Makes the brute-forcing quadratically more complex in practice.

Reply:
Aha. I think I see it. Because you maybe could brute force two values at the same time?